src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -8,17 +9,24 @@ namespace NHibernate.Tool.hbm2ddl
 	// Candidate to be exstensions of ISessionFactory and Configuration
 	public static class SchemaMetadataUpdater
 	{
-		public static void Update(ISessionFactory sessionFactory)
+		public static void Update(ISessionFactoryImplementor sessionFactory)
 		{
-			var factory = (ISessionFactoryImplementor) sessionFactory;
-			var dialect = factory.Dialect;
-			var connectionHelper = new SuppliedConnectionProviderConnectionHelper(factory.ConnectionProvider);
-			factory.Dialect.Keywords.UnionWith(GetReservedWords(dialect, connectionHelper));
+			var dialect = sessionFactory.Dialect;
+			var connectionHelper = new SuppliedConnectionProviderConnectionHelper(sessionFactory.ConnectionProvider);
+			sessionFactory.Dialect.Keywords.UnionWith(GetReservedWords(dialect, connectionHelper));
 		}
 
 		public static void QuoteTableAndColumns(Configuration configuration)
 		{
-			ISet<string> reservedDb = GetReservedWords(configuration.GetDerivedProperties());
+			// Instatiates a new instance of the dialect so doesn't benefit from the Update call.
+			var dialect = Dialect.Dialect.GetDialect(configuration.GetDerivedProperties());
+			QuoteTableAndColumns(configuration, dialect);
```

Comment:
Ok, thanks for the feedback.

I reverted the original signature for `QuoteTableAndColumns` to it's previous behavior, marked it obsolete, and nothing in NHibernate now uses it.  Everywhere that calls the new `QuoteTableAndColumns` overload calls `Update` beforehand (except for the two explicit unit tests).

I also refactored the `GetReservedWords` to better encapsulate the dispose pattern they are using (finally { Release() }).